### PR TITLE
t-test on test data

### PR DESCRIPTION
**Pull request by sophieball**
Perform t-test on test data rather than training data. The t-stats are qualitatively the same (same direction) but some of them are not significant on the held-out test data.

### CODE REVIEW

**Comment by CaptainEmerson** [comment]
Sophie, do you want me to run this as well? If so, what target do you want me to run, and what output do you need?

**Comment by sophieball** [comment]
Right! Should've mentioned.
Please run it on both pushback and linguistic datasets. The target is `train_classifier_g` and the outputs should be the 2 `.log` files. Thanks~

**Comment by sophieball** [comment]
Also, @CaptainEmerson , when I was plotting feature importance, I noticed that the Google results do not have some politeness strategies, such as, apologizing, btw.. I used to drop them in `convo_politeness.py` but not anymore.. When you run the current code, can you check if `Apologizing` is among the features? Just a search in the `.log` file is sufficient.

File: main/get_feature_set.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 drop_cols = ["Indirect_(btw)", "Indirect_(greeting)",
   "Apologizing", "Deference",
   "SUBJUNCTIVE", "INDICATIVE"]
-text_based = list(set(text_based) - set(drop_cols))
+#text_based = list(set(text_based) - set(drop_cols))
```
Comment (CaptainEmerson) [inline main/get_feature_set.py:21]: rm?

File: main/train_classifier_g.py [hunk 2]
```diff
@@ -56,33 +56,41 @@ def train_model(training_data, model_name="svm", pretrain=False, what_data="issu
   elif model_name == "rf":
     s.set_model_function(classifiers.random_forest_model)
     # RF params
-    s.add_parameter("n_estimators",
-                    [int(x) for x in np.linspace(start=200, stop=2000, num=10)])
-    s.add_parameter("max_features", ["auto", "sqrt"])
-    s.add_parameter("max_depth", [int(x) for x in np.linspace(10, 110, num=11)])
+    s.set_parameters({
+        "n_estimators": [int(x) for x in np.linspace(start=10, stop=100, num=10)],
```
Comment (CaptainEmerson) [inline main/train_classifier_g.py:60]: it's generally best to separate pull requests into small pieces. for instance, I'd put this parameter tweaking into its own one. if this set_parameters refactoring is more than just these 4 lines, I'd also have a separate refactoring PR

File: src/convo_politeness.py [hunk 3]
```diff
@@ -202,7 +202,7 @@ def train_polite(comments):
 # input: a pandas dataframe: _id, text
 # output: a pandas dataframe: _id, text, politeness
 def get_politeness_score(comments):
-  logging.info("Beging calculating politeness, rows: {}".format(len(comments)))
+  logging.info("Begining calculating politeness, rows: {}".format(len(comments)))
```
Comment (CaptainEmerson) [inline src/convo_politeness.py:205]: nit: Beginning, an example of Muphry's Law

File: src/suite.py [hunk 4]
```diff
@@ -224,16 +228,16 @@ def remove_SE_comment(features_df, Google, row, model, features, max_values, tf_
...
+      #if not Google:
+      #  # print out some quotes for debugging
+      #  logging.info("going to be flipped:{}, {}: {}".format(row["thread_id"],
+      #          row["label"], text))
+      #  logging.info("values:")
+      #  new_row = new_features.iloc[0]
+      #  logging.info(("\n".join([str((k, row[k], new_row[k])) for k in new_features.columns])))
+      #  logging.info("after being flipped: word:{},  new sentence: {}".format(word, 
+      #          new_sentence))
+      #  logging.info("\n")
```
Comment (CaptainEmerson) [inline src/suite.py:240]: rm all this?

File: main/train_classifier_g.py (hunk 2 above)
Comment (CaptainEmerson) [reply main/train_classifier_g.py:60]: I won't insist on this now, since we're in a hurry

Comment (sophieball) [reply main/train_classifier_g.py:60]: Got it~

File: src/suite.py (hunk 4 above)
Comment (sophieball) [reply src/suite.py]: done.

File: src/convo_politeness.py (hunk 3 above)
Comment (sophieball) [reply src/convo_politeness.py]: fixed :p

File: main/get_feature_set.py (hunk 1 above)
Comment (sophieball) [reply main/get_feature_set.py]: done

**Comment by CaptainEmerson** [comment]
```
Traceback (most recent call last):
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/main/train_classifier_g.runfiles/__main__/main/train_classifier_g.py", line 16, in <module>
    from src import suite
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/main/train_classifier_g.runfiles/__main__/src/suite.py", line 21, in <module>
    from src import convo_politeness
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/main/train_classifier_g.runfiles/__main__/src/convo_politeness.py", line 37, in <module>
    f = open("src/data/speakers_bots_full.list")
FileNotFoundError: [Errno 2] No such file or directory: 'src/data/speakers_bots_full.list'
Traceback (most recent call last):
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/src/convo_word_freq_diff.runfiles/__main__/src/convo_word_freq_diff.py", line 9, in <module>
    from convokit import Corpus, Speaker, Utterance
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/src/convo_word_freq_diff.runfiles/deps/pypi__convokit/convokit/__init__.py", line 4, in <module>
    from .politenessStrategies import *
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/src/convo_word_freq_diff.runfiles/deps/pypi__convokit/convokit/politenessStrategies/__init__.py", line 1, in <module>
    from .politenessStrategies import *
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/src/convo_word_freq_diff.runfiles/deps/pypi__convokit/convokit/politenessStrategies/politenessStrategies.py", line 5, in <module>
    from convokit.text_processing.textParser import process_text
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/src/convo_word_freq_diff.runfiles/deps/pypi__convokit/convokit/text_processing/__init__.py", line 2, in <module>
    from .textParser import *
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/src/convo_word_freq_diff.runfiles/deps/pypi__convokit/convokit/text_processing/textParser.py", line 2, in <module>
    import spacy
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/src/convo_word_freq_diff.runfiles/deps/pypi__spacy/spacy/__init__.py", line 10, in <module>
    from thinc.api import prefer_gpu, require_gpu, require_cpu  # noqa: F401
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/src/convo_word_freq_diff.runfiles/deps/pypi__thinc/thinc/api.py", line 6, in <module>
    from .model import Model, serialize_attr, deserialize_attr
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/src/convo_word_freq_diff.runfiles/deps/pypi__thinc/thinc/model.py", line 13, in <module>
    from .shims import Shim
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/src/convo_word_freq_diff.runfiles/deps/pypi__thinc/thinc/shims/__init__.py", line 2, in <module>
    from .pytorch import PyTorchShim
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/src/convo_word_freq_diff.runfiles/deps/pypi__thinc/thinc/shims/pytorch.py", line 18, in <module>
    from .pytorch_grad_scaler import PyTorchGradScaler
ModuleNotFoundError: No module named 'thinc.shims.pytorch_grad_scaler'
There were 22 warnings (use warnings() to see them)

```

**Comment by sophieball** [comment]
I tried the current commit in a new directory.. It should work.. No rush, though. This new output won't be too different from the previous one

**Comment by sophieball** [comment]
> Right! Should've mentioned. Please run it on both pushback and linguistic datasets. The target is `train_classifier_g` and the outputs should be the 2 `.log` files, the 3 `roc_curve_*.png` files, and `features_xxxx.csv`. Thanks~

Added a ROC curve per Bogdan's suggestion. There will be 3 `.png' files.

**Comment by CaptainEmerson** [comment]
```
Traceback (most recent call last):
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/src/convo_word_freq_diff.runfiles/__main__/src/convo_word_freq_diff.py", line 9, in <module>
    from convokit import Corpus, Speaker, Utterance
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/src/convo_word_freq_diff.runfiles/deps/pypi__convokit/convokit/__init__.py", line 4, in <module>
    from .politenessStrategies import *
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/src/convo_word_freq_diff.runfiles/deps/pypi__convokit/convokit/politenessStrategies/__init__.py", line 1, in <module>
    from .politenessStrategies import *
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/src/convo_word_freq_diff.runfiles/deps/pypi__convokit/convokit/politenessStrategies/politenessStrategies.py", line 5, in <module>
    from convokit.text_processing.textParser import process_text
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/src/convo_word_freq_diff.runfiles/deps/pypi__convokit/convokit/text_processing/__init__.py", line 2, in <module>
    from .textParser import *
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/src/convo_word_freq_diff.runfiles/deps/pypi__convokit/convokit/text_processing/textParser.py", line 2, in <module>
    import spacy
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/src/convo_word_freq_diff.runfiles/deps/pypi__spacy/spacy/__init__.py", line 10, in <module>
    from thinc.api import prefer_gpu, require_gpu, require_cpu  # noqa: F401
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/src/convo_word_freq_diff.runfiles/deps/pypi__thinc/thinc/api.py", line 6, in <module>
    from .model import Model, serialize_attr, deserialize_attr
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/src/convo_word_freq_diff.runfiles/deps/pypi__thinc/thinc/model.py", line 13, in <module>
    from .shims import Shim
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/src/convo_word_freq_diff.runfiles/deps/pypi__thinc/thinc/shims/__init__.py", line 2, in <module>
    from .pytorch import PyTorchShim
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/src/convo_word_freq_diff.runfiles/deps/pypi__thinc/thinc/shims/pytorch.py", line 18, in <module>
    from .pytorch_grad_scaler import PyTorchGradScaler
ModuleNotFoundError: No module named 'thinc.shims.pytorch_grad_scaler'

```

**Comment by sophieball** [comment]
Probably because we previously limited thinc's version. It is in the newest version: https://github.com/explosion/thinc/blob/master/thinc/shims/pytorch_grad_scaler.py

I removed the constraint

**Comment by sophieball** [comment]
:/
I force it to be the latest version..

**Comment by sophieball** [comment]
Did you try `bazel clean` then rebuild?

**Comment by CaptainEmerson** [comment]
The dependency issue is fixed, it looks like. But I get:
```
[nltk_data]   Package averaged_perceptron_tagger is already up-to-
[nltk_data]       date!
Log saved in `/usr/local/google/home/emersonm/toxicity-detector/train_classifier.log`
sh: line 1: bazel-bin/src/convo_word_freq_diff: No such file or directory
```

I'm running from here:
`setwd("~/toxicity-detector")`

**Comment by sophieball** [comment]
The difference is that t-tests show that adjusting for SE words degrades the results.. so I'm reporting results before adjustment.. otherwise I don't know how to defend why we do the adjustment

**Comment by sophieball** [comment]
Yes, it's under feed_data

```
r_binary(
    name = "feed_data",
    src = "feed_data.R",
    data = [
        ":train_classifier_g",
        ":train_polite_score",
        "//src:convo_word_freq_diff",
        "//src:find_SE_words",
        "//main:train_prompt_types",
        "//main:train_polite_prompt_classifier",
    ],
    deps = [
        ":politeness_logi",
        "@R_plyr",
        "@R_readr",
    ],
)
```

**Comment by CaptainEmerson** [comment]
But train_classifier_g doesn't depend on feed_data or convo_word_freq_diff, right?

**Comment by CaptainEmerson** [comment]
Ah, you are right, I think that the dependencies are fine. Do you need new results from convo_word_freq_diff?

**Comment by CaptainEmerson** [comment]
The pngs are getting overwritten on each run, right? I did a run yesterday/last night, and it ran both the regular version and then the linguistic version. So while I have the log files from both runs, I guess the PNGs I have are only the linguistic ones.

**Comment by sophieball** [comment]
no. just the most recent version on both datasets.
I'm changing the logging.
Removed `to_csv`

**Comment by sophieball** [comment]
@CaptainEmerson Can I merge this?

**Comment by CaptainEmerson** [comment]
yep, sg

**Pull request merged**